api/api/spec/spec.yaml

```diff
@@ -7605,8 +7615,6 @@ paths:
           $ref: '#/components/responses/UnauthorizedResponse'
         '403':
           $ref: '#/components/responses/PermissionDeniedResponse'
-        '404':
-          $ref: '#/components/responses/ResourceNotFoundResponse'
```

Comment:
I would keep previous references to `#/components/responses/ResourceNotFoundResponse`, since there are some endpoints where 404 is actually returned:
```JSON
# DELETE /agents/group?agents_list=000&group_id=test
{
  "title": "Resource Not Found",
  "detail": "The group does not exist",
  "type": "about:blank",
  "remediation": "Please, use `GET /agents/groups` to find all available groups",
  "dapi_errors": {
    "master-node": {
      "error": "The group does not exist"
    }
  },
  "error": 1710
}
``` 

What I meant [here](https://github.com/wazuh/wazuh/pull/20428#discussion_r1494407416) was to remove only the new 404 codes introduced in this PR, or in other words, only `InvalidUriResponse`.

Reply:
I Reverted all the removed lines found in the `4.9.`0 branch like the following:
```
        '404':
          $ref: '#/components/responses/ResourceNotFoundResponse'
```

Reply:
I removed the lines from the `spec.yaml` file with the `'404'` response code that I've added unnecessarily.